apps/web/utils/useUniversitySearch.tsx

```diff
@@ -0,0 +1,45 @@
+import Fuse from 'fuse.js'
+
+interface FilterProps {
+  key: string
+  value: Array<string>
+}
+
+interface SearchProductsProps {
+  fuseInstance: Fuse<any>
+  query?: string
+  activeFilters: Array<FilterProps>
+}
+
+interface QueryMakerProps {
+  $and: Array<any>
+}
+export const SearchProducts = ({
```

Comment:
Just curious: Do I understand correctly that this runs client-side in memory?
So this is a fuzzy search library for a moderately sized dataset?

Reply:
yes that is correct